features/keychain/module/keychain.js

```diff
@@ -44,30 +44,51 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
-  #assertPrivateKeysUnlocked() {
-    assert(!this.#privateKeysAreLocked, 'private keys are locked')
+  #assertPrivateKeysUnlocked(seedIds) {
+    const locked = this.#checkPrivateKeysLocked(seedIds)
+    assert(!locked, 'private keys are locked')
   }
 
-  arePrivateKeysLocked() {
-    return this.#privateKeysAreLocked
+  #checkPrivateKeysLocked(seedIds) {
+    if (!seedIds?.length) {
+      return Object.values(this.#seedLockStatus).some(Boolean)
+    }
+
+    return seedIds.some((seedId) => {
+      assert(
+        Object.hasOwn(this.#seedLockStatus, seedId),
+        `cannot check lock state for unknown seed "${seedId}"`
+      )
+      return this.#seedLockStatus[seedId]
+    })
+  }
+
+  arePrivateKeysLocked(seeds = []) {
```

Comment:
Not a fan of passing in the seeds here. But I assume that only happens once when the second seed is first added? Or why are we not passing in a seed id in the first place?